packages/assertion-monitor/index.ts

```diff
@@ -63,6 +60,34 @@ async function getValidatorWhitelistDisabled(
   return contract.read.validatorWhitelistDisabled()
 }
 
+function calculateSearchWindow(
+  childChainInfo: ChainInfo,
+  parentChain: ReturnType<typeof getChainFromId>
+): { days: number; blocks: number } {
+  const blockTime = getBlockTimeForChain(parentChain)
+  const initialBlocksToSearch =
+    childChainInfo.confirmPeriodBlocks * VALIDATOR_AFK_BLOCKS
+  const timespan = blockTime * initialBlocksToSearch
+
+  const blocksInDays = timespan / (60 * 60 * 24)
+  const blocksInDaysMinusSafety = Math.max(blocksInDays - SAFETY_BUFFER_DAYS, 0)
+  const daysAdjustedForMax = Math.min(
+    Math.ceil(blocksInDaysMinusSafety),
+    MAXIMUM_SEARCH_DAYS
+  )
+
+  // Calculate the maximum number of blocks for 7 days
+  const maxBlocksFor7Days = Math.floor((7 * 24 * 60 * 60) / blockTime)
```

Comment:
Would it make sense to decouple it from number 7 and have it as maxSearchableBlocks having 7 replaced by MAX_SEARCHABLE_DAYS?